activation/handler.go

```diff
@@ -165,93 +125,84 @@ func (h *Handler) SyntacticallyValidate(ctx context.Context, atx *types.Activati
 		if atx.InitialPost == nil {
 			return errors.New("no prev atx declared, but initial post is not included")
 		}
-		if atx.InnerActivationTx.NodeID == nil {
+		if atx.NodeID == nil {
 			return errors.New("no prev atx declared, but node id is missing")
 		}
 		if atx.VRFNonce == nil {
 			return errors.New("no prev atx declared, but vrf nonce is missing")
 		}
-		if atx.CommitmentATX == nil {
+		if atx.CommitmentATXID == nil {
 			return errors.New("no prev atx declared, but commitment atx is missing")
 		}
-		if *atx.CommitmentATX == types.EmptyATXID {
+		if *atx.CommitmentATXID == types.EmptyATXID {
 			return errors.New("empty commitment atx")
 		}
 		if atx.Sequence != 0 {
 			return errors.New("no prev atx declared, but sequence number not zero")
 		}
 
-		// Use the NIPost's Post metadata, while overriding the challenge to a zero challenge,
-		// as expected from the initial Post.
-		initialPostMetadata := *atx.NIPost.PostMetadata
-		initialPostMetadata.Challenge = shared.ZeroChallenge
 		if err := h.nipostValidator.VRFNonce(
-			atx.SmesherID, *atx.CommitmentATX, atx.VRFNonce, &initialPostMetadata, atx.NumUnits,
+			atx.SmesherID, *atx.CommitmentATXID, *atx.VRFNonce, atx.NumUnits,
 		); err != nil {
 			return fmt.Errorf("invalid vrf nonce: %w", err)
 		}
+		post := wire.PostFromWireV1(atx.InitialPost)
 		if err := h.nipostValidator.Post(
-			ctx, atx.SmesherID, *atx.CommitmentATX, atx.InitialPost, &initialPostMetadata, atx.NumUnits,
+			ctx, atx.SmesherID, *atx.CommitmentATXID, post, shared.ZeroChallenge, atx.NumUnits,
 		); err != nil {
 			return fmt.Errorf("invalid initial post: %w", err)
 		}
```

Comment:
It makes sense as `PostV2` would probably be a different or updated algorithm. So far, because `types.Post ==  wire.PostV1`, I did this translation to reduce the PR size.

Reply:
This will need to change when we add ATX v2